pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/RebalanceResult.java

```diff
@@ -29,23 +29,31 @@
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class RebalanceResult {
+  private final String _rebalanceId;
   private final Status _status;
   private final Map<InstancePartitionsType, InstancePartitions> _instanceAssignment;
   private final Map<String, InstancePartitions> _tierInstanceAssignment;
-  private final Map<String, Map<String, String>> _segmentAssignment;
+  private final Map<String, Map<String, String>> _targetSegmentAssignment;
   private final String _description;
 
   @JsonCreator
-  public RebalanceResult(@JsonProperty(value = "status", required = true) Status status,
+  public RebalanceResult(@JsonProperty(value = "rebalanceId", required = true) String rebalanceId,
+      @JsonProperty(value = "status", required = true) Status status,
       @JsonProperty(value = "description", required = true) String description,
       @JsonProperty("instanceAssignment") @Nullable Map<InstancePartitionsType, InstancePartitions> instanceAssignment,
       @JsonProperty("tierInstanceAssignment") @Nullable Map<String, InstancePartitions> tierInstanceAssignment,
-      @JsonProperty("segmentAssignment") @Nullable Map<String, Map<String, String>> segmentAssignment) {
+      @JsonProperty("targetSegmentAssignment") @Nullable Map<String, Map<String, String>> targetSegmentAssignment) {
```

Comment:
Let's not change this name because it is incompatible with the current response format (imagine client code look for this field)